src/main/c/cext/ruby.c

```diff
@@ -76,8 +76,26 @@ void rb_check_frozen(VALUE object) {
   truffle_invoke(RUBY_CEXT, "rb_check_frozen", object);
 }
 
+void rb_insecure_operation(void) {
+  rb_raise(rb_eSecurityError, "Insecure operation: -r");
```

Comment:
What does the `-r` part means? Is that from MRI?

Reply:
Not sure what that means. Yes, from MRI.

Reply:
https://github.com/ruby/ruby/blob/463a56d14b9241c8631d71d286760ea092d70c36/safe.c#L103-L114
So it tries to get the caller method name and fallback to `-r` if there is no caller.